pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -444,6 +455,34 @@ func updateVMIsPhase(nodeName string, vmis []*k6tv1.VirtualMachineInstance, ch c
 	}
 }
 
+func updateVMIEvictionBlocker(nodeName string, vmis []*k6tv1.VirtualMachineInstance, ch chan<- prometheus.Metric) {
+	for _, vmi := range vmis {
+		mv, err := prometheus.NewConstMetric(
+			vmiEvictionBlockerDesc, prometheus.GaugeValue,
+			checkNonEvictableVMAndSetMetric(vmi),
+			nodeName, vmi.Namespace, vmi.Name,
+		)
+		if err != nil {
+			continue
+		}
+		ch <- mv
+	}
+}
+
+func checkNonEvictableVMAndSetMetric(vmi *k6tv1.VirtualMachineInstance) float64 {
+	setVal := 0.0
+	if vmi.IsEvictable() {
+		vmiIsMigratableCond := controller.NewVirtualMachineInstanceConditionManager().
+			GetCondition(vmi, k6tv1.VirtualMachineInstanceIsMigratable)
+
+		if vmiIsMigratableCond != nil && vmiIsMigratableCond.Status == k8sv1.ConditionFalse {
+			setVal = 1.0
+		}
```

Comment:
Should this predicate also include " `|| vmiIsMigratableCond == nil` " ? Or in non-unit-test scenario, the `vmiIsMigratableCond` will not be `nil` ? 

It looks like that some code path will calculate the live-migration condition. For instance, at https://github.com/kubevirt/kubevirt/blob/27d138b38d368af4a72a1d5162e8012eb272c228/pkg/virt-handler/vm.go#L999 . 

So maybe if the `vmiIsMigratableCond` is `nil`, the migratable condition needs to be calculated, or the check function needs to wait a while?

Reply:
I think the vmiIsMigratableCond  can't be null at this  point, however I think it won't harm to add this condition case as well . Will change it.

Reply:
> Should this predicate also include " `|| vmiIsMigratableCond == nil` " ? Or in non-unit-test scenario, the `vmiIsMigratableCond` will not be `nil` ?

I think the vmiIsMigratableCond can't be null at this point, however I think it won't harm to add this condition case as well . Will change it.